Extension/src/LanguageServer/configurations.ts

```diff
@@ -988,7 +988,7 @@ export class CppProperties {
             } else {
                 // However, if compileCommands are used and compilerPath is explicitly set, it's still necessary to resolve variables in it.
                 if (configuration.compilerPath === "${default}") {
-                    configuration.compilerPath = settings.defaultCompilerPath;
+                    configuration.compilerPath = settings.defaultCompilerPath ?? undefined;
                 }
                 if (configuration.compilerPath === null) {
```

Comment:
What's up with this check? It seems impossible to hit? compilerPath is not typed to be null.

![image](https://github.com/user-attachments/assets/65733a95-cc07-402e-9da1-fed1f2930117)

Should `configuration.compilerPathIsExplicit = true;` be set in the block above if ${default} is used and defaultCompilerPath is null?

Reply:
`settings.defaultCompilerPath` used to return string or undefined. I was just attempting to maintain the current behavior. We should mark this as a follow up item since we're trying not to change behavior in this PR.